mesonpy/__init__.py

```diff
@@ -187,14 +203,32 @@ def basename(self) -> str:
             version=self._project.version,
         )
 
+    @property
+    def tags(self) -> packaging.tags.Tag:
+        """Wheel tags."""
+        if self.is_pure:
+            return packaging.tags.Tag('py3', 'none', 'any')
+        # Get the most specific tag for the Python interpreter.
+        tag = next(packaging.tags.sys_tags())
+        if tag.platform.startswith('manylinux'):
+            tag = packaging.tags.Tag(tag.interpreter, tag.abi, _adjust_manylinux_tag(tag.platform))
```

Comment:
I don't want to claim that what my patch does is beautiful, but what you propose unfortunately does not work. `InterpreterTag.python` is unfortunately not complete as reported in #121. If we depend on `packaging` anyway, I would delegate to that for computing the interpreter tag correctly. Also, as you can see from the code of `packaging` or `wheel` or the code in `meson-python` now, `sysconfig.get_platform()` does not always return the right thing as you need to check for a 32-bit Python running on a 64-bit kernel.

All in all, I think the if we want to delegate some work to `packaging` it makes sense to delegate as much of it as possible.

Reply:
I don't think there is value in splitting this into separate functions and it would require either internal caching or calling multiple times possibly expensive `packaging.tags` functions. We need the tag only in a couple of places and always in its complete form. Anyhow, the reason why I decided to return a `packaging.tags.Tag` instead of a string is indeed that this way accessing the individual parts of the tag is straightforward.

Reply:
> I don't want to claim that what my patch does is beautiful, but what you propose unfortunately does not work. `InterpreterTag.python` is unfortunately not complete as reported in #121. If we depend on `packaging` anyway, I would delegate to that for computing the interpreter tag correctly.

You are right, I opened https://github.com/pypa/packaging/pull/612. We can copy the `interpreter_tag` implementation from https://github.com/pypa/packaging/pull/613 for now.

> Also, as you can see from the code of `packaging` or `wheel` or the code in `meson-python` now, `sysconfig.get_platform()` does not always return the right thing as you need to check for a 32-bit Python running on a 64-bit kernel.

For now we can use the first value from `linux_platforms`, but I think this functionality should either be provided by `packaging` directly or `sysconfig`.

Reply:
> but I think this functionality should either be provided by packaging directly or sysconfig.

I've actually been thinking about this quite a bit recently, because Meson cannot depend on `packaging` or `setuptools` (and still uses distutils for more than just Debian installation schemes). I think the appropriate location is sysconfig, because it's information about the current python interpreter as it is running on the current system, and is basically set in stone as a result, whereas `packaging.tags` (in theme with the general approach of `packaging`) is about the set of all python interpreters intersecting all OSes, including the ones released after the pair that you're currently running.

Speaking of which, I heard you can do something about that now. :)

Reply:
I guess what I'd like to see is something like `sysconfig.platform_tag(stable=False, pure=True)` and then it could tell you what abi standard the current interpreter uses, and infer e.g. the most recent manylinux/musllinux tag based on the loaded glibc or musl.

Reply:
> You are right, I opened [pypa/packaging#612](https://github.com/pypa/packaging/issues/612). We can copy the `interpreter_tag` implementation from [pypa/packaging#613](https://github.com/pypa/packaging/pull/613) for now.

I can do that but I don't see the advantage. Why calling that function is better than calling `sys_tags()` and extracting the component we are interested in? I understand that the API provided by `packaging` is not the most effective butI think the suboptimal API is better than cooking our own half baked version.

> For now we can use the last value from `linux_platforms`, but I think this functionality should either be provided by `packaging` directly or `sysconfig`.

There is the same problem on macOS. But again, I don't see why getting the thing piecewise is better than calling one function be done with it.